guides/source/action_controller_overview.md

```diff
@@ -666,6 +668,128 @@ end
 
 You may notice in the above code that we're using `render xml: @users`, not `render xml: @users.to_xml`. If the object is not a String, then Rails will automatically invoke `to_xml` for us.
 
+The default serializers that control this behavior are defined in <tt>ActionController::Renderers</tt>.
+
+You may customize how +XML+, +JSON+, or +JS+ are formatted using the +serializing+ macro.
+In any controller, pass in a hash to +serializing+ where each element is a +format+ to a callable that
+takes two arguments: the +object+ to be serialized and any +options+ passed to +render+.
+
+Below, we customize +JSON+ serialization for the <tt>UsersController</tt> and its children to pretty-print +JSON+.
+
+```diff
+class UsersController < ApplicationController
++   serializing json: ->(json, options) do
++     if json.is_a?(String)
++       json
++     else
++       json = json.as_json(options) if json.respond_to?(:as_json)
++       JSON.pretty_generate(json, options)
++     end
++   end
+
+  def index
+    @users = User.all
+    respond_to do |format|
+      format.html # index.html.erb
+      format.xml  { render xml: @users}
+      format.json { render json: @users}
+    end
+  end
+end
+```
+
+There are two other ways we could have customized +JSON+ serialization in the <tt>UsersController</tt>.
+In previous versions of Rails, these were the only ways.
+
+One, by knowing that the default Renderer does not modify strings, we could have changed the code as follows:
+
+```diff
+-      format.json { render json: @users}
++      format.json { render json: JSON.pretty_generate(@users.as_json) }
+```
+
+Note that this would only affect the +#index+ action.
+
+Two, by knowing that when the +JSON+ renderer is added (<tt>ActionController::Renderers.add :json, &block</tt>),
```

Comment:
This is talking about an implementation detail and something that we consider private API, so I think it is better to not add this section to the documentation.